### DISCUSSION TITLE
Error: Packed type exceeds maximum width

### DISCUSSION THREAD

**Discussion by hankhsu1996** [MikePopoloski/slang#1180] [Q&A] [answered]
I encountered the following error when running slang:

```
error: packed type of 16777216 bits is too large; maximum width is 16777215 bits
```

Our design is large, and we are wondering if there is a way to configure slang to handle wider packed types or increase the maximum width limit. Any guidance or suggestions would be appreciated!

**Comment by MikePopoloski** [comment, marked as answer]
It's certainly unusual to need a packed object that large; typically you switch to unpacked well before that limit. What size do you need supported? And can you say which tool(s) you're running on that support such a size? I believe Xcelium, for example, doesn't support packed sizes larger than that. The LRM says:
```
The maximum size of a packed array can be limited, but shall be at least 65 536 (2^16) bits.
```
So that's probably all you can rely on if you want to be portable. That being said, there's probably room to bump slang's limit up a bit, though it would be good to know what the target is.

**Comment by hankhsu1996** [comment]
I only hit the limit temporarily, and at this point, I don’t think the target is set in stone. The simulator I’m using is VCS.

For paranoid reasons, I primarily use packed arrays because, historically, the synthesizer had better support for packed signals than unpacked signals.

I wasn’t aware of this size limit in the LRM—thank you for pointing it out. Since this is specified by the LRM, I think the current implementation is fine. I’ll likely either change the signal to unpacked or break it down into smaller pieces.
